server/grpc_service.go

```diff
@@ -908,3 +908,81 @@ func (s *Server) incompatibleVersion(tag string) *pdpb.ResponseHeader {
 		Message: msg,
 	})
 }
+
+// SyncMaxTS is a RPC method used to synchronize the timestamp of TSO between the
+// Global TSO Allocator and multi Local TSO Allocator leaders. It contains two
+// phases:
+// 1. Collect timestamps among all Local TSO Allocator leaders, and choose the
+//    greatest one as MaxTS.
+// 2. Send the MaxTS to all Local TSO Allocator leaders. They will compare MaxTS
+//    with its current TSO in memory to make sure their local TSOs are not less
+//    than MaxTS by writing MaxTS into memory to finish the global TSO synchronization.
+func (s *Server) SyncMaxTS(ctx context.Context, request *pdpb.SyncMaxTSRequest) (*pdpb.SyncMaxTSResponse, error) {
+	if err := s.validateInternalRequest(request.GetHeader()); err != nil {
+		return nil, err
+	}
+	tsoAllocatorManager := s.GetTSOAllocatorManager()
+	// Get all Local TSO Allocator leaders
+	allocatorLeaders, err := tsoAllocatorManager.GetLocalAllocatorLeaders()
+	if err != nil {
+		return nil, err
+	}
+	processedDCs := make([]string, 0)
+	if request.GetMaxTs() == nil || request.GetMaxTs().Physical == 0 {
+		// The first phase of synchronization: collect the max local ts
+		var maxLocalTS pdpb.Timestamp
+		for _, allocator := range allocatorLeaders {
+			if !allocator.IsStillAllocatorLeader() {
+				return nil, errs.ErrNotLocalAllocatorLeader.FastGenByArgs(
```

Comment:
can just skip it here because global allocator will check if all DCs are handled.